platform/http_client.cpp

```diff
@@ -0,0 +1,190 @@
+#include "platform/http_client.hpp"
+
+#include "base/string_utils.hpp"
+
+#include "std/sstream.hpp"
+
+namespace platform
+{
+HttpClient & HttpClient::SetDebugMode(bool debug_mode)
+{
+  m_debugMode = debug_mode;
+  return *this;
+}
+
+HttpClient & HttpClient::SetUrlRequested(string const & url)
+{
+  m_urlRequested = url;
+  return *this;
+}
+
+HttpClient & HttpClient::SetHttpMethod(string const & method)
+{
+  m_httpMethod = method;
+  return *this;
+}
+
+HttpClient & HttpClient::SetBodyFile(string const & body_file,
+                           string const & content_type,
+                           string const & http_method /* = "POST" */,
+                           string const & content_encoding /* = "" */)
+{
+  m_bodyFile = body_file;
+  m_bodyData.clear();
+  m_contentType = content_type;
+  m_httpMethod = http_method;
+  m_contentEncoding = content_encoding;
+  return *this;
+}
+
+HttpClient & HttpClient::SetReceivedFile(string const & received_file)
+{
+  m_receivedFile = received_file;
+  return *this;
+}
+
+HttpClient & HttpClient::SetUserAgent(string const & user_agent)
+{
+  m_userAgent = user_agent;
+  return *this;
+}
+
+HttpClient & HttpClient::SetBodyData(string const & body_data,
+                           string const & content_type,
+                           string const  & http_method /* = "POST" */,
+                           string const & content_encoding /* = "" */)
+{
+  m_bodyData = body_data;
+  m_bodyFile.clear();
+  m_contentType = content_type;
+  m_httpMethod = http_method;
+  m_contentEncoding = content_encoding;
+  return *this;
+}
+
+HttpClient & HttpClient::SetBodyData(string && body_data,
```

Comment:
Make this method template, receive body data by a universal reference. Otherwise, create a common method that inits m_contentType, clears m_bodyFile, etc...